### PR TITLE
bugfix: CLDSRV-11 Remove ownership checks

### PR DESCRIPTION
**Pull request by miniscruff**
This should also resolve the integration test

### CODE REVIEW

**Review by alexanderchan-scality — DISMISSED** [review]

File: lib/api/objectDelete.js [hunk 1]
```diff
@@ -22,10 +22,6 @@ const versionIdUtils = versioning.VersionID;
  */
 function objectDelete(authInfo, request, log, cb) {
     log.debug('processing request', { method: 'objectDelete' });
-    if (authInfo.isRequesterPublicUser()) {
```
Comment (naren-rajendran) [inline lib/api/objectDelete.js]: I am still not sure that object delete can be performed by non-owners (public, authenticated and other account access), but based on the behavior we saw, this should be ok.

File: lib/api/objectDelete.js [hunk 2]
```diff
@@ -22,10 +22,6 @@ const versionIdUtils = versioning.VersionID;
...
-        log.debug('operation not available for public user');
-        return cb(errors.AccessDenied);
-    }
```
Comment (alexanderchan-scality) [inline lib/api/objectDelete.js]: should this be addressed in a different ticket (to make cloudserver more like aws)? it seems to be out of scope of the fix, and I believe this may be something that would warrant further discussion.

Comment (alexanderchan-scality) [reply lib/api/objectDelete.js]: would be undoing a part of a related commit that has more to it: https://github.com/scality/cloudserver/commit/12b866dc680df403bb771a59426e9e9e18b101bd

Comment (miniscruff) [reply lib/api/objectDelete.js]: Those 4ish checks seem like things that should be checked in the ACL stuff tbh. I can undo that work and tweak the tests so that the delete does not work anymore and save the rest for later.

File: lib/api/objectDelete.js (hunk 1 above)
Comment (miniscruff) [reply lib/api/objectDelete.js]: Will undo this change and adjust the tests accordingly

**Review by naren-rajendran — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]
